### PR TITLE
AEA-1381 Enable config forwarding to error handler and decision maker handler

### PR DESCRIPTION
**Pull request by DavidMinarsch**
Make error handler instance rather than class based

## Proposed changes

The PR makes error handler and decision maker handler configurable. In particular, config object defined in yaml can be forwarded to it.

It also makes error handler instance based to avoid cross-agent effects for multiple agents within a given process.

## Fixes

- #1977, #2300

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

-

### CODE REVIEW

File: aea/decision_maker/default.py [hunk 1]
```diff
@@ -83,14 +83,13 @@ def role_from_first_message(  # pylint: disable=unused-argument
     signing_dialogue_class = SigningDialogue
     signing_msg_class = SigningMessage
 
-    def __init__(self, identity: Identity, wallet: Wallet) -> None:
```
Comment (DavidMinarsch) [inline aea/decision_maker/default.py:86]: need to filter configs in base so they're not set on simple name space!

File: aea/aea_builder.py [hunk 2]
```diff
@@ -1589,11 +1623,14 @@ def set_from_configuration(
...
             file_path = agent_configuration.decision_maker_handler["file_path"]
+            config = agent_configuration.error_handler["config"]
             self.set_decision_maker_handler_paths(dotted_path, file_path)
+            self.set_decision_maker_handler_config(config)
         if agent_configuration.error_handler != {}:
             dotted_path = agent_configuration.error_handler["dotted_path"]
             file_path = agent_configuration.error_handler["file_path"]
-            self.set_error_handler(dotted_path, file_path)
+            config = agent_configuration.error_handler["config"]
+            self.set_error_handler(dotted_path, file_path, config)
```
Comment (marcofavorito) [inline aea/aea_builder.py:1633]: I wonder why we have `set_decision_maker_handler_paths` and `set_decision_maker_handler_config` but for handler we have only `set_error_handler` (maybe it is trivial, I haven't checked)

**Review by marcofavorito — APPROVED** [review]
LGTM

File: aea/aea_builder.py (hunk 2 above)
Comment (DavidMinarsch) [reply aea/aea_builder.py:1633]: updated

File: aea/decision_maker/default.py (hunk 1 above)
Comment (DavidMinarsch) [reply aea/decision_maker/default.py:86]: updated

File: aea/aea_builder.py (hunk 2 above)
Comment (marcofavorito) [reply aea/aea_builder.py:1633]: ok

Comment (DavidMinarsch) [reply aea/aea_builder.py:1633]: (the decision maker approach is the correct one as it guarantees loading happens **after** the protocols, contracts and connections are loaded, so error handler can depend on them (as it should be able to))

File: .pylintrc [hunk 3]
```diff
@@ -34,7 +34,7 @@ max-locals=31
 max-statements=80
 max-parents=11
 max-branches=24
-max-attributes=34
+max-attributes=36
```
Comment (marcofavorito) [inline .pylintrc:37]: instead of this, disable exception to "AEABuilder"?

Comment (solarw) [reply .pylintrc:37]: actually AEABuilder is overbloated, lets keep max-attribtes low to get remind about attributes limit

**Review by solarw — APPROVED** [review]
LGTM

**Review by marcofavorito — APPROVED** [review]
LGTM

**Pull request merged**